pkg/diagnostics/diagnostics.go

```diff
@@ -115,7 +115,7 @@ func (m *DiagnosticsManager) getModuleStatus(ctx context.Context, module string)
 	}
 }
 
-func (m *DiagnosticsManager) isOnline(ctx context.Context) bool {
+func (m *DiagnosticsManager) isHealthy() bool {
```

Comment:
The idea of diagnostics to return realtime health of system modules. I am wondering if we need to return health also since health checks results can also be retrieved over the perf api which is already exposed

Reply:
Maybe it's better approach to have diagnostics unrelated to healthchecks. Unless there is something else i am missing

Reply:
it was asked to add the network state for the diagnostics call for a future plan to use this call in the proxy or clients to decide if the node is reachable or not.
since we merged the checks (network/cache) i thought it is better to check for both and decide the total node health instead of just the network. 

so in the `proxy/node-indexer` the diagnostics call will be used to check for both fields `SystemOk` and `Healthy` to decide if the node is healthy or not